tests/e2e/e2e_os_test.go

```diff
@@ -0,0 +1,139 @@
+package e2e
+
+import (
+	"bytes"
+	"fmt"
+	"github.com/pkg/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"os/exec"
+	"strconv"
+	"strings"
+	"testing"
+)
+
+func runCmd(cmdS string) ([]byte, error) {
+	var cmd *exec.Cmd
+	var out, stdErr bytes.Buffer
+	cmd = exec.Command("/bin/sh", "-c", cmdS)
+
+	cmd.Stdout = &out
+	cmd.Stderr = &stdErr
+
+	err := cmd.Run()
+	if err != nil {
+		return nil, fmt.Errorf("error running command %v: %s", cmd, err)
+
+	}
+	return out.Bytes(), nil
+}
+
+func waitForBuildComplete(namespace string, buildName string) error {
+	// Interval of 5
+	return wait.Poll(5, jobTimeout, func() (bool, error) {
+		var buildOut bool
+		buildStatus, err := runCmd("oc get build --namespace=" + namespace + " --template='{{ range .items }}{{ if (eq .metadata.name \"" +
```

Comment:
Maybe add a TODO to this function indicating that this will be eventually replaced by an API? Not too sure.. Maybe we could keep it as using runCmd.